src/services/keycloakService.ts

```diff
@@ -0,0 +1,122 @@
+import * as WebConfigAPI from '../services/WebConfigService';
+declare global {
+  interface Window {
+    Keycloak?: any;
+  }
+}
+/**
+ * Enum with possible authentication roles in keycloak
+ *
+ */
+export enum AUTH_ROLE {
+  Admin = 'admin',
+  User = 'user',
+  System = 'system-user',
+  Power = 'power-user',
+}
+
+/**
+ * Class managing authentication functionality
+ */
+class KeycloakService {
+  //We can't get KeycloakInstance type because of dynamic loading of keycloak library
+  private keycloakAuth: any;
+
+  private isKeycloakInitialized;
+
+  constructor() {
+    this.isKeycloakInitialized = this.init();
+  }
+
+  /**
+   * Initialize keycloak and create instance
+   *
+   * @returns Promise
+   */
+  private init(): Promise<any> {
+    const keycloakConfig = WebConfigAPI.getWebConfig().keycloak;
+
+    this.keycloakAuth = window.Keycloak({
+      url: keycloakConfig.url,
+      realm: keycloakConfig.realm,
+      clientId: keycloakConfig.clientId,
+    });
+
+    return new Promise((resolve, reject) => {
+      this.keycloakAuth
+        .init({ onLoad: 'check-sso', checkLoginIframeInterval: 1 })
+        .then(() => {
+          resolve('success');
+        })
+        .catch((errorData: any) => {
+          reject(errorData);
+        });
+    });
+  }
+  /**
+   * Returns promise of keycloak initialization
+   *
+   * @returns Promise
+   */
+  public isInitialized(): Promise<any> {
+    return this.isKeycloakInitialized;
+  }
+  /**
+   * Returns if user is authenticated
+   *
+   * @returns boolean
+   */
+  public isAuthenticated(): boolean {
+    return this.keycloakAuth.authenticated!;
+  }
+
+  /**
+   * Initiate login process in keycloak
+   *
+   * @returns Promiseb
+   */
+  public login(): Promise<any> {
+    return this.keycloakAuth.login();
+  }
+
+  /**
+   * Initiate logout process in keycloak
+   *
+   * @param redirectUri uri to redirect after logout
+   */
+  public logout(redirectUri?: String): void {
+    this.keycloakAuth.logout({ redirectUri });
+  }
+
+  /**
+   * Gets keycloak token
+   *
+   * @returns String
+   */
+  public getToken(): String {
+    return this.keycloakAuth.token;
+  }
+
+  /**
+   * Gets user name from keycloak
+   *
+   * @returns String
```

Comment:
what value is returned when the user is not logged in? `returns` should be focused on values. Otherwise other developers can not be sure AppLayout.tsx:71 (and other occurrences) will be evaluated properly.